source/common/http/http2/codec_impl.cc

```diff
@@ -62,7 +62,7 @@ ConnectionImpl::StreamImpl::StreamImpl(ConnectionImpl& parent, uint32_t buffer_l
   }
 }
 
-ConnectionImpl::StreamImpl::~StreamImpl() { ASSERT(unconsumed_bytes_ == 0); }
```

Comment:
Q: What's the case in which this assert no longer holds true? nit: Can you delete empty destructor?